tdp/core/models/deployment_model.py

```diff
@@ -181,27 +182,59 @@ def from_dag(
                 and operation.name.action == "restart"
                 and len(operation.playbook.hosts) > 0
             )
-            deployment.operations.append(
-                OperationModel(
-                    operation=operation.name.name,
-                    operation_order=operation_order,
-                    host=None,
-                    extra_vars=None,
-                    state=OperationStateEnum.PLANNED,
-                )
-            )
-            operation_order += 1
-            if can_perform_rolling_restart:
+            if host_name:
+                for host in host_name:
+                    if (
+                        isinstance(operation, PlaybookOperation)
+                        and host in operation.playbook.hosts
+                    ):
+                        deployment.operations.append(
+                            OperationModel(
+                                operation=operation.name.name,
+                                operation_order=operation_order,
+                                host=host,
+                                extra_vars=None,
+                                state=OperationStateEnum.PLANNED,
+                            )
+                        )
+                        operation_order += 1
+                        if can_perform_rolling_restart:
+                            deployment.operations.append(
+                                OperationModel(
+                                    operation=OPERATION_SLEEP_NAME,
+                                    operation_order=operation_order,
+                                    host=None,
+                                    extra_vars=[
+                                        f"{OPERATION_SLEEP_VARIABLE}={rolling_interval}"
+                                    ],
+                                    state=OperationStateEnum.PLANNED,
+                                )
+                            )
+                    operation_order += 1
```

Comment:
Are you sure about the indentation ? We should increment `operation_order` only if the sleep operation is added no ?